app/lib/api/file_system/file_system_dav.dart

```diff
@@ -199,11 +199,20 @@ class DavRemoteDocumentFileSystem extends DocumentRemoteSystem {
   @override
   Future<void> updateFile(String path, List<int> data,
       {bool forceSync = false}) async {
-    // Create directory if not exists
-    final directoryPath = p.dirname(path);
-    final directory = Directory(directoryPath);
-    if (!await directory.exists()) {
-      await directory.create(recursive: true);
+    // Cache check
+    if (!forceSync && remote.hasDocumentCached(path)) {
+      cacheContent(path, data);
+      return;
```

Comment:
This was a mistake on my side, we shouldn't return when it should be cached, it should also update it on the server

Reply:
okay, so you want to go back to this? 
`Future<void> updateFile(String path, List<int> data,
    {bool forceSync = false}) async {
  if (!forceSync && remote.hasDocumentCached(path)) {
    cacheContent(path, data);
  }`

Reply:
Just remove the return line and it should work just fine